tools/common/StaticRegistrar.cs

```diff
@@ -3690,6 +3677,14 @@ void Generate (string header_path, string source_path)
 							hdr.WriteLine ("#include <objc/runtime.h>");
 							hdr.WriteLine ("#include <objc/message.h>");
 
+#if MONOMAC
+							// 32-bit header files do not define these types
+							if (!Is64Bits) {
```

Comment:
So the only reason I did these was to reduce the delta that could break customers (indirectly, was not sure if really could happen).

Before extensions, when we added those types, we would registers all types, including those 64-bit only on 32-bit builds. Adding the extension APIs broke this, since the types aren't even in the header files. Forward declaring these two let us register that mass of 64-bit only types even in 32-bit builds still..

Not registering 64-bit only types would obviously be cleaner, if you think it is safe @rolfbjarne. XM has the luxury of being less backwards compatible that XI.